### PR TITLE
Watchable Prefs

### PR DESCRIPTION
**Pull request by gausie**
Screenshot 2022-11-17 at 16 05 27

This PR gives the ability to watch certain prefs from a panel in the GUI. This is great for tracking certain state values as a run goes on, or tracking the ongoing results for a running script.

We might want to add a way for scripts to add to this panel without having to do a pref write eventually. I also might make a cell flash when it changes. But those will come in a later PR

### CODE REVIEW

File: src/net/sourceforge/kolmafia/preferences/Preferences.java [hunk 1]
```diff
@@ -762,6 +762,10 @@ private static void encodeCharacter(char ch) {
                         : "\\u" + Integer.toHexString(ch);
   }
 
+  public static boolean propertyExists(final String name) {
+    return propertyExists(name, true) || propertyExists(name, false);
+  }
```
Comment (jaadams5) [inline src/net/sourceforge/kolmafia/preferences/Preferences.java:767]: This looks weird.  I might refactor so there were userPropertyExists(name) and globalPropertyExists(name) for readability but this is not a blocker nor even a firm well thought out preference.

**Review by jaadams5 — DISMISSED** [review]
I was looking for a gui that added and removed preferences to be watched.  Not there.  Need a hint to the operator to set using the command line but that can wait for the next phase..  I also think I'd like to be able to pop this out as a separate window so I can still have my Daily Deeds.  That may already be possible but I couldn't figure it out.  Regardless  it is an interesting idea, at least until someone wants to know what line in a script triggered the change or worse yet, enter a (non-existent) debugger at the change.  :-)

**Comment by gausie** [comment]
> I was looking for a gui that added and removed preferences to be watched. Not there. Need a hint to the operator to set using the command line but that can wait for the next phase.. I also think I'd like to be able to pop this out as a separate window so I can still have my Daily Deeds. That may already be possible but I couldn't figure it out. Regardless it is an interesting idea, at least until someone wants to know what line in a script triggered the change or worse yet, enter a (non-existent) debugger at the change. :-)

There is a context menu you can use to add and remove values! But now I realise that that context menu might not work in all cases so I'll make a few changes to this PR

**Comment by Veracity0** [comment]
Did you break context menus in the ItemManager?

The first three items in my General (i.e. Inventory) panel

1) "DRINK ME" potion
2) "KICK ME" sign
3) 11-leaf clover

- Go to General
- Select 11-leaf clover
- Right Click to get context menu
- Select Game Description
- Get confused when KoLmafia shows the item description for the "DRINK ME" potion

**Comment by gausie** [comment]
> Did you break context menus in the ItemManager?
> 
> The first three items in my General (i.e. Inventory) panel
> 
>     1. "DRINK ME" potion
> 
>     2. "KICK ME" sign
> 
>     3. 11-leaf clover
> 
> 
>     * Go to General
> 
>     * Select 11-leaf clover
> 
>     * Right Click to get context menu
> 
>     * Select Game Description
> 
>     * Get confused when KoLmafia shows the item description for the "DRINK ME" potion

Huh. Should be an easy fix, I'll sort that

**Pull request merged**